internal/services/network/public_ips_data_source.go

```diff
@@ -99,8 +117,18 @@ func dataSourcePublicIPsRead(d *pluginsdk.ResourceData, meta interface{}) error
 			}
 		}
 
+		attachmentStatus, attachmentStatusOk := d.GetOk("attachment_status")
```

Comment:
It's possible for `attachmentStatus` here to be nil, which could cause a panic in the type assertion.  Perhaps set this to either `""` or `All` prior to `GetOk()` to guard against this?

Reply:
Solved by checking the `attachmentStatusOk`